js/src/currency-converter.js

```diff
@@ -0,0 +1,376 @@
+
+    if (user_inputs.year < 1975) {
+        buying_power_money_value = user_inputs.pounds + (user_inputs.shillings / 20) + (user_inputs.old_pence / 240);
+
+        if(user_inputs.pounds > 0) {
+            bp_string = "£" + user_inputs.pounds;
+        }
+        if(user_inputs.shillings > 0){
+            if(user_inputs.pounds > 0) {
+                bp_string = bp_string + ", ";
+            }
+
+            bp_string = bp_string + user_inputs.shillings + "s";
+        }
+        if(user_inputs.old_pence > 0) {
+            if(user_inputs.pounds > 0 || user_inputs.shillings > 0){
+                bp_string = bp_string +" & ";
+            }
+            bp_string = bp_string + user_inputs.old_pence + "d";
+        }
+
+        /* Displays the currency in the buying power box, e.g. £10 1s 0d would be 10 pounds, 1 shilling, 0 pence.
+        bp_string = "£" + user_inputs.pounds + ", " + user_inputs.shillings + "s & " + user_inputs.old_pence + "d";
+        */
+    }
+    else {
+        buying_power_money_value = user_inputs.pounds + user_inputs.new_pence;
+
+        // Since it will be modern currency (after 1975) we can use toLocaleString to show the money value.
+        bp_string = buying_power_money_value.toLocaleString('en-GB', {style: 'currency', currency: 'GBP'});
+    }
+
+    return {
+        horses: Math.floor(buying_power_money_value / get_horse_price(user_inputs.year)),
+        cows: Math.floor(buying_power_money_value / get_cow_price(user_inputs.year)),
+        wool: Math.floor(buying_power_money_value / get_wool_price(user_inputs.year)),
+        wheat: Math.floor(buying_power_money_value / get_wheat_price(user_inputs.year)),
+        wage: Math.floor(buying_power_money_value / get_wage_price(user_inputs.year)),
+        money: currency_money_to_modern_value,
+        bp_string: bp_string
+    };
+
+
+}
+
+function get_inflation_rate(year) {
+    return conversion_data[year].inflation;
+}
+
+function get_horse_price(year) {
+    return conversion_data[year].horse_price;
+}
+
+function get_cow_price(year) {
+    return conversion_data[year].cow_price;
+}
+
+function get_wool_price(year) {
+    return conversion_data[year].wool_price;
+}
+
+function get_wheat_price(year) {
+    return conversion_data[year].wheat_price;
+}
+
+function get_wage_price(year) {
+    return conversion_data[year].wage_price;
+}
+
+function get_century(year) {
+    if (year >= 2000) {
+        return "21st";
+    }
+    else if (year >= 1900) {
+        return "20th";
+    }
+    else if (year >= 1800) {
+        return "19th";
+    }
+    else if (year >= 1700) {
+        return "18th";
+    }
+    else if (year >= 1600) {
+        return "17th";
+    }
+    else if (year >= 1500) {
+        return "16th";
+    }
+    else if (year >= 1400) {
+        return "15th";
+    }
+    else if (year >= 1300) {
+        return "14th";
+    }
+    else if (year >= 1200) {
+        return "13th";
+    }
+}
+
+function get_currency_year() {
```

Comment:
Looking at these `get_currency` functions it seems they're pretty much doing the same thing. I wonder whether these are actually one function that takes a string and returns an integer. For `get_currency_new_pence()` you could have the division inline.

Reply:
Hi Gwyn,

Should they be in one function and then have a string parameter with the jQuery ID instead?

Reply:
Hi @AshTNA - that would be one way of doing it. You'd then end up with calls like 

```javascript
get_currency('#currency-shillings');
```
Another alternative might be to accept a string that you use within the function to determine which element to obtain, for example: 

```javascript
get('shillings');
```
Which do you prefer?